### PR TITLE
Streetnetwork: remove allocs around dijkstra

### PR DESCRIPTION
**Pull request by pbougue**

### CODE REVIEW

**Review by TeXitoi — APPROVED** [review]

**Comment by kinnou02** [comment]
Allocation during street network are fully removed,
On test request (a little bit extreme) we go down to 470ms from 1019ms.
Almost no time is spent in kernel space and call to clear_page have been eliminated

File: source/georef/dijkstra_shortest_paths_with_heap.h [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+Stay tuned using
+twitter @navitia
+IRC #navitia on freenode
+https://groups.google.com/d/forum/navitia
+www.navitia.io
+*/
+
+#include <boost/graph/dijkstra_shortest_paths.hpp>
+
```
Comment (kinnou02) [inline source/georef/dijkstra_shortest_paths_with_heap.h:32]: a little comment about what we did that will be nice I think
> :white_check_mark: Done

File: source/georef/street_network.h [hunk 2]
```diff
@@ -168,20 +174,25 @@ struct PathFinder {
     template<class Visitor>
     void dijkstra(vertex_t start, Visitor visitor) {
         // Note: the predecessors have been updated in init
-        boost::two_bit_color_map<> color(boost::num_vertices(geo_ref.graph));
+
+        // Fill color map in white before dijkstra
+        std::fill(color.data.get(),
```
Comment (kinnou02) [inline source/georef/street_network.h:179]: why don't we do this in the init?

**Review by kinnou02 — APPROVED** [review]

File: source/georef/street_network.h (hunk 2 above)
Comment (pbougue) [reply source/georef/street_network.h:179]: We do 2 dijkstra calls (one on each node of the edge receiving projection).

**Pull request merged**